src/hooks/useUploadsManager/useUploadsManager.tsx

```diff
@@ -1,62 +1,212 @@
+  const navigate = useNavigate()
   const { uploadsState, addAsset, updateAsset } = useUploadsManagerStore()
+  const { displaySnackbar } = useSnackbar()
   const [uploadsProgress, setUploadsProgress] = useState<UploadsProgressRecord>({})
+  const [assetsFiles, setAssetsFiles] = useState<AssetFile[]>([])
+  const { activeChannelId } = useUser()
+  const { channel, loading: channelLoading } = useChannel(activeChannelId ?? '')
+  const { videos, loading: videosLoading } = useVideos(
+    {
+      where: {
+        id_in: uploadsState.filter((item) => item.parentObject.type === 'video').map((item) => item.parentObject.id),
+      },
+    },
+    { skip: !uploadsState.length }
+  )
+
+  const uploadsStateWithProgress: AssetUploadWithProgress[] = uploadsState.map((asset) => ({
+    ...asset,
+    progress: uploadsProgress[asset.contentId] ?? 0,
+  }))
+
+  const channelDataObjects = [channel?.avatarPhotoDataObject, channel?.coverPhotoDataObject]
+  const videosDataObjects = videos?.flatMap((video) => [video.mediaDataObject, video.thumbnailPhotoDataObject]) || []
+  const allDataObjects = [...channelDataObjects, ...videosDataObjects]
+
+  // Enriching data with pending/accepted/rejected status
+  const uploadsStateWithLiaisonJudgement = uploadsStateWithProgress
+    .map((asset) => {
+      const dataObject = allDataObjects.find((dataObject) => dataObject?.joystreamContentId === asset.contentId)
+      if (!dataObject && !channelLoading && !videosLoading) {
+        return null
+      }
+
+      return { ...asset, liaisonJudgement: dataObject?.liaisonJudgement, ipfsContentId: dataObject?.ipfsContentId }
+    })
+    .filter((asset) => asset !== null)
+
+  const lostConnectionAssets = uploadsStateWithLiaisonJudgement.filter(
+    (asset) =>
+      asset?.liaisonJudgement === LiaisonJudgement.Pending &&
+      asset.lastStatus === 'inProgress' &&
+      asset.progress === 0 &&
+      !assetsFiles.find((item) => item.contentId === asset.ipfsContentId)
+  )
+
+  useEffect(() => {
+    if (!lostConnectionAssets.length) {
+      return
+    }
+    displaySnackbar({
+      title: `(${lostConnectionAssets.length}) Asset${
+        lostConnectionAssets.length > 1 ? 's' : ''
+      } waiting to resume upload`,
+      description: 'Reconnect files to fix the issue',
+      actionText: 'See assets',
+      onActionClick: () => navigate(absoluteRoutes.studio.uploads()),
+      iconType: 'warning',
+    })
+  }, [displaySnackbar, lostConnectionAssets.length, navigate])
+
+  // Enriching video type assets with video title
+  const uploadsStateWithVideoTitles = uploadsStateWithLiaisonJudgement.map((asset) => {
+    if (asset?.type === 'video') {
+      const video = videos?.find((video) => video.mediaDataObject?.joystreamContentId === asset.contentId)
+      const title = video?.title ?? null
+      return { ...asset, title }
+    }
+    return asset
+  })
+
+  // Check if liaison data and video title is available
+  uploadsStateWithVideoTitles.map((asset) => {
```

Comment:
In general, in cases like this where you don't care about return value but just want to iterate over the values, `.forEach` may be a better choice. It's a bit more performant, although negligibly, but mostly just for readability improvement

No need to change here